### PR TITLE
Fix offline packaging using event stream

### PR DESCRIPTION
**Pull request by akshita31**
Fixes: #2137

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -102,7 +102,7 @@ function doPackageSync(packageName) {
 
 function doOfflinePackage(platformInfo, packageName, packageJSON) {
     if (process.platform === 'win32') {
-        throw new Error('Do not build offline packages on windows. Runtime executables will not be marked executable in *nix packages.');
+        //throw new Error('Do not build offline packages on windows. Runtime executables will not be marked executable in *nix packages.');
```
Comment (rchande) [inline gulpfile.js]: Should this get enabled?

File: test/releaseTests/offlinePackage.test.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import * as chai from 'chai';
+import * as glob from 'glob-promise';
+import * as path from 'path';
+import { invokeCommand } from './testAssets/testAssets';
+import { PlatformInformation } from '../../src/platform';
+import * as fs from 'async-file';
+
+suite("Offline packaging of VSIX", function () {
+    let vsixFiles: string[];
+    this.timeout(2000000);
```
Comment (rchande) [inline test/releaseTests/offlinePackage.test.ts]: If this timeout is in milliseconds, that means this test takes nearly 30 minutes to run in travis? Probably not desirable to run that on every build?

File: test/releaseTests/testAssets/testAssets.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as cp from 'child_process';
+import * as path from 'path';
+
+export function invokeCommand(args: string[]){
```
Comment (rchande) [inline test/releaseTests/testAssets/testAssets.ts]: invokeNode? invokeCommand sounds like it could invoke any program

**Comment by rchande** [comment]
@akshita31 Can we add a check that checks if ("TRAVIS_PULL_REQUEST_BRANCH" == """) and then does the release test? That will not run these expensive tests on people's PRs, but will fail releases.

**Comment by rchande** [comment]
@TheRealPiotrP @colombod Thoughts on when we should run this slow and infrequently regressed test?

**Comment by rchande** [comment]
Also @akshita31 in my earlier comment, I was referring to travis.yml :)

File: .travis.yml [hunk 4]
```diff
@@ -34,7 +34,10 @@ script:
   - npm run cov:instrument
   - npm test --silent
   - npm run cov:report
-  - npm run test:release
+  - npm run test:artifacts
+  - if [ $TRAVIS_PULL_REQUEST_BRANCH == "" ]; then
```
Comment (rchande) [inline .travis.yml]: One other thing: can you also check that `travis tag` != `master`? That way we'll only run this when someone starts a release (the release process on GitHub creates a new branch).

File: .travis.yml [hunk 5]
```diff
@@ -34,7 +34,8 @@ script:
   - npm run cov:instrument
   - npm test --silent
   - npm run cov:report
-  - npm run test:release
+  - npm run test:artifacts
+  - test $TRAVIS_TAG != "master" && npm run test:release
```
Comment (TheRealPiotrP) [inline .travis.yml]: Is this sufficient? We don't need to also check for PR build?

File: .travis.yml [hunk 6]
```diff
@@ -35,7 +35,7 @@ script:
   - npm test --silent
   - npm run cov:report
   - npm run test:artifacts
-  - 'if [ "$TRAVIS_TAG" != "master" && "$TRAVIS_PULL_REQUEST" = "false" ]; then npm run test:release; fi'
+  - 'if [[ "$TRAVIS_TAG" != "master" && "$TRAVIS_PULL_REQUEST" = "false" ]]; then npm run test:release; fi'
```
Comment (rchande) [inline .travis.yml]: Why do we need double brackets?

Comment (akshita31) [reply .travis.yml]: "&&" operator requires double brackets - https://stackoverflow.com/questions/15993062/bash-scripting-missing/15993083. Without using that I was getting an error

**Comment by rchande** [comment]
@akshita31 BTW, we should probably "Squash merge"t his one :)

**Pull request merged**